### PR TITLE
Ensure NEG Readiness Reflector commits endpoints from non-default subnets.

### PR DESCRIPTION
**Pull request by sawsa307**
/assign @gauravkghildiyal

### CODE REVIEW

File: pkg/neg/syncers/transaction_test.go [hunk 1]
```diff
@@ -839,196 +839,357 @@ func TestFilterEndpointByTransaction(t *testing.T) {
...
+	vals := gce.DefaultTestClusterValues()
+	vals.SubnetworkURL = defaultTestSubnetURL
+	_, transactionSyncer := newTestTransactionSyncer(negtypes.NewAdapter(gce.NewFakeGCECloud(vals)), negtypes.VmIpPortEndpointType, false)
+	reflector := &testReflector{}
+	transactionSyncer.reflector = reflector
+
+	syncerKey := transactionSyncer.NegSyncerKey
+	negName := transactionSyncer.NegName
+	prevFlag := flags.F.EnableMultiSubnetClusterPhase1
+	defer func() { flags.F.EnableMultiSubnetClusterPhase1 = prevFlag }()
```
Comment (gauravkghildiyal) [inline pkg/neg/syncers/transaction_test.go:851]: Nice thought about parallel run tests and resetting flags

File: pkg/neg/syncers/transaction_test.go [hunk 2]
```diff
@@ -1038,12 +1199,12 @@ func TestCommitPods(t *testing.T) {
...
 		negNameSet := sets.NewString(reflector.negNames...)
-		if len(expectOutput) != 0 && !(negNameSet.Len() == 1 && negNameSet.Has(transactionSyncer.NegSyncerKey.NegName)) {
-			t.Errorf("For test case %q, expect neg name to be %v, but got %v", tc.desc, transactionSyncer.NegSyncerKey.NegName, negNameSet.List())
+		if len(expectOutput) != 0 && negNameSet.Len() != 2 {
+			t.Errorf("For test case %q, expect two negs, but got %v", tc.desc, negNameSet.List())
 		}
 
-		if !reflect.DeepEqual(expectOutput, reflector.endpointMaps) {
-			t.Errorf("For test case %q, expect endpoint map to be %v, but got %v", tc.desc, expectOutput, reflector.endpointMaps)
+		if !reflect.DeepEqual(expectOutput, reflector.pollMap) {
```
Comment (gauravkghildiyal) [inline pkg/neg/syncers/transaction_test.go:1206]: nit: Let's use cmp.Diff 

https://google.github.io/styleguide/go/decisions#equality-comparison-and-diffs

File: pkg/neg/readiness/poller.go [hunk 3]
```diff
@@ -45,16 +45,16 @@ const (
 	hcRetryDelay = time.Second
 )
 
-// negMeta references a GCE NEG resource
-type negMeta struct {
+// NegMeta references a GCE NEG resource
+type NegMeta struct {
```
Comment (gauravkghildiyal) [inline pkg/neg/readiness/poller.go:49]: Trying to understand the need to export this. Is this only motivated by the usage within the test of `syncers` package? If yes, I'd argue the `fakeReflector` defined within the `syncers` package should also define it's own type which happens to include these three same parameters. That way, our unit tests don't dictate visibility of structs of other packages.

**Review by gauravkghildiyal — COMMENTED** [review]
Minor comments but looking good overall. Thanks for the work.

File: pkg/neg/readiness/poller.go (hunk 3 above)
Comment (sawsa307) [reply pkg/neg/readiness/poller.go:49]: Sounds good, updated.

File: pkg/neg/syncers/transaction.go [hunk 4]
```diff
@@ -735,9 +735,27 @@ func (s *transactionSyncer) commitPods(endpointMap map[negtypes.EndpointGroupInf
...
+		syncerKey := s.NegSyncerKey
+		if flags.F.EnableMultiSubnetClusterPhase1 {
+			defaultSubnet, err := utils.KeyName(s.networkInfo.SubnetworkURL)
+			if err != nil {
+				s.logger.Error(err, "Errored getting default subnet from NetworkInfo when committing pods")
+				continue
+			}
+
+			if endpointGroupInfo.Subnet != defaultSubnet {
+				negName, err = s.getNonDefaultSubnetName(endpointGroupInfo.Subnet)
```
[the remark this answers is not in the document]
Comment (sawsa307) [reply pkg/neg/syncers/transaction.go:748]: Sounds good, updated.

File: pkg/neg/syncers/transaction.go [hunk 5]
```diff
@@ -735,9 +735,27 @@ func (s *transactionSyncer) commitPods(endpointMap map[negtypes.EndpointGroupInf
...
+				if err != nil {
+					s.logger.Error(err, "Errored getting non-default subnet NEG name when committing pods")
+					continue
+				}
+			}
+			// To ensure syncerKey has the same information as the passed in NEG name.
+			syncerKey.NegName = negName
+		}
 		// Only zone is needed because NEGs from non-default subnet have different names.
```
[the remark this answers is not in the document]
Comment (sawsa307) [reply pkg/neg/syncers/transaction.go:757]: Removed, thanks!

File: pkg/neg/syncers/transaction_test.go (hunk 2 above)
Comment (sawsa307) [reply pkg/neg/syncers/transaction_test.go:1206]: Updated.

**Review by sawsa307 — COMMENTED** [review]
Addressed all comments, PTAL! @gauravkghildiyal

**Comment by gauravkghildiyal** [comment]
/lgtm

Thanks!

**Pull request merged**